### ISSUE TITLE
keras.ops.numpy.i0 overflows to Inf for finite float64 result near x=713

### ISSUE THREAD

**Issue by ALinrunrun** [keras-team/keras#23400]
## Description

`keras.ops.numpy.i0` returns `inf` for an input whose correct `float64` result is still finite.

For the input:

```text
713.0
```

the modified Bessel function `i0(713.0)` should be approximately:

```text
6.705128263670996e+307
```

This value is very large, but it is still within the finite `float64` range. However, `keras.ops.numpy.i0` returns `inf`.

This looks like an incorrect overflow / numerical accuracy issue in `keras.ops.numpy.i0` for large finite `float64` inputs.

## Minimal reproduction

```python
import os
os.environ["KERAS_BACKEND"] = "tensorflow"
os.environ["CUDA_VISIBLE_DEVICES"] = "-1"
os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
os.environ["TF_ENABLE_ONEDNN_OPTS"] = "0"

import warnings
warnings.filterwarnings("ignore")

import numpy as np
import keras

x = np.array([713.0], dtype=np.float64)

actual = np.asarray(
    keras.ops.numpy.i0(
        keras.ops.convert_to_tensor(x, dtype="float64"),
    ),
    dtype=np.float64,
)

expected = np.array([6.705128263670996e+307], dtype=np.float64)

print("keras:", keras.__version__)
print("backend:", keras.config.backend())
print("input:", x.tolist())
print("actual:", actual.tolist())
print("expected:", expected.tolist())
```

Output:

```text
keras: 3.14.0
backend: tensorflow
input: [713.0]
actual: [inf]
expected: [6.705128263670996e+307]
```

## Expected behavior

`keras.ops.numpy.i0` should return a finite `float64` value close to:

```text
[6.705128263670996e+307]
```

The expected value is based on a high-precision mpmath reference rounded to `float64`. Although the result is close to the upper end of the `float64` range, it is still finite.

## Actual behavior

`keras.ops.numpy.i0` currently returns:

```text
[inf]
```

The result incorrectly overflows to infinity even though the correct `float64` value is finite.

**Comment by maitry63** [comment]
Hi @ALinrunrun , Thank you for reporting this issue! But I think this issue should be open in the Tensorflow not in Keras.

I checked into the implementation for the TensorFlow backend. The implementation in `keras/src/backend/tensorflow/numpy.py` is:

```python
def i0(x):
    x = convert_to_tensor(x)
    dtype = standardize_dtype(x.dtype)
    if dtype in ["int64", "float64"]:
        dtype = "float64"
    elif dtype not in ["bfloat16", "float16"]:
        dtype = config.floatx()
    x = tf.cast(x, dtype)
    return tf.math.bessel_i0(x)
```
This shows:
- Converts the input to a TensorFlow tensor.
- Preserves float64 inputs , while other floating-point types follow Keras' configured default dtype.
- Delegates the actual computation directly to `tf.math.bessel_i0`.

I also verified using TensorFlow directly:

```python
import tensorflow as tf

x = tf.constant([713.0], dtype=tf.float64)
print(tf.math.bessel_i0(x).numpy())
```
Output:
```text
[inf]
```
Check this [gist](https://colab.sandbox.google.com/gist/maitry63/8d106b9467a3fabca262ba9d80e9419f/keras-ops-numpy-i0-overflows-to-inf-for-finite-float64-result-near-x-713-23400.ipynb#scrollTo=yu6WACVvvVy_).

This reproduces the same result without using` Keras`. Since `tf.math.bessel_i0` also returns `inf `for the same `float64` input, the issue comes from TensorFlow's implementation. The `Keras` wrapper only passes the input to TensorFlow, so it is not causing the overflow.

From this Observation, let me know, If we can close the issue here & please try to reopen this in `Tensorflow` Repo.

**Comment by ALinrunrun** [comment]
Hi @maitry63, thanks for checking this.

I agree with your analysis. On the TensorFlow backend, keras.ops.numpy.i0 just calls tf.math.bessel_i0, and the same 713.0 already returns inf in TensorFlow, so this overflow is not introduced by Keras.

I'll file an issue on TensorFlow and link it here. Fine to close this Keras issue.

Thanks!
